src/content/resources/glossary.md

```diff
@@ -336,3 +336,51 @@ class A implements B {} // A is a subtype of B, but NOT a subclass of B.
 
 class C extends D {} // C is a subtype AND a subclass of D.
 ```
+
+## Variance and variance positions
+
+A type parameter of a class (or a mixin, etc) is said to be _covariant_
+when the type as a whole "co-varies" with the actual type argument, that
+is: if we replace the type argument by a subtype then the type as a whole
+is also a subtype.
+
+For example, the type parameter of the class `List` is covariant because
+list types co-vary with their type argument: `List<int>` is a subtype of
+`List<Object>` because `int` is a subtype of `Object`.
+
+In Dart, all type parameters of all class, mixin, mixin class, and enum
+declarations are covariant.
+
+However, function types are different: A function type is covariant in the
+return type, but the opposite (known as _contravariant_) in its parameter
+types.  For example, the type `int Function(int)` is a subtype of the type
+`Object Function(int)`, but it is a supertype of `int Function(Object)`.
+
+This makes sense if you consider their substitutability (see the section
+about subtypes): If you expect at compile time to work with a function of
+type `int Function(int)` then it is OK to actually work with a function of
+type `int Function(Object)` at run time: You will pass an `int` to it (and
+that's fine, it accepts any `Object`), and it returns an `int` (which is
+exactly what you'd expect). Hence, `int Function(Object)` is a subtype of
+`int Function(int)`. Note that everything is turned upside-down when we
+consider a parameter type.
```

Comment:
By "turned upside-down" do you mean "reversed" (as in the opposite) and what does "everything" here mean? The subtype relationship described in the preceding prose or the logic followed?

Reply:
The point is that if `X` occurs in a covariant position in `T` then `X` occurs in a contravariant position in `void Function(T)`, and if `Y` occurs in a contravariant position in `T` then `Y` occurs in a covariant position in `void Function(T)`. Finally, if `Z` occurs in an invariant position in `T` then it occurs in an invariant position in `void Function(T)`.

That's rather long and tedious, so I just said "everything" is reversed.

I don't know if it's a good idea to elaborate on this (to make this particular point explicit) or it's better to leave it as imprecise as it is currently. I suspect that the nested positions are much, much rarer than the basic ones (so people will write and/or encounter types like `void Function(T)`, but they are much less likely to encounter `void Function(void Function(T))`). If this is true then they won't ever need to use this "reverse everything" rule.

I expanded a bit on the text here.